src/movies.js

```diff
@@ -1,22 +1,74 @@
 // Iteration 1: All directors? - Get the array of all directors.
 // _Bonus_: It seems some of the directors had directed multiple movies so they will pop up multiple times in the array of directors.
+
+const movies = require("./data")
+
 // How could you "clean" a bit this array and make it unified (without duplicates)?
-function getAllDirectors() {}
+function getAllDirectors(movies) {
+  const directors = movies.map (function (movie) {
+    return movie.director
+  })
+  return directors
+}
+getAllDirectors(movies)
 
 // Iteration 2: Steven Spielberg. The best? - How many drama movies did STEVEN SPIELBERG direct?
-function howManyMovies() {}
+function howManyMovies(movies) {
+  const spielbergMovies = movies.filter (function(movie) {
+    if (movie.director === 'Steven Spielberg' && movie.genre.includes('Drama')) {
+      return true;
+    }
+  })
+  
+return spielbergMovies.length
+
+}
 
 // Iteration 3: All scores average - Get the average of all scores with 2 decimals
-function scoresAverage() {}
+function scoresAverage(movies) {
+  if (movies.length === 0) return 0;
+  const average = movies.reduce(function(sum, movie) {
+    if (movie.score === '' || movie.score === undefined) {
+      movie.score = 0;
+    }
+    return sum + movie.score
+  }, 0)
+   average / movies.length;
```

Comment:
This line doesn't do anything right now. If you want to reassign the value to average, you need to first declare it with the ```let``` folder and then do ```average /= movies.length ``` which stands for ```average = average / movies.length```/

The other option would be to divide it already in the reduce:
```
const average = movies.reduce(function(sum, movie) {
    if (movie.score === '' || movie.score === undefined) {
      movie.score = 0;
    }
    return sum + movie.score
  }, 0) / movies.length;
```